### PR TITLE
Add support for running netopeer2-server in foreground with syslog

### PR DESCRIPTION
**Pull request by troglobit**
This patch adds support for running `netopeer2-server` under init systems like [Finit](https://github.com/troglobit/finit/), which supervise processes running in the foreground.  Similar to `-d`, but still using syslog.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -1119,7 +1120,7 @@ main(int argc, char *argv[])
 {
     int ret = EXIT_SUCCESS;
     int c, *idx, i;
-    int daemonize = 1, verb = 0;
+    int daemonize = 1, verb = 0, use_syslog = 0;
```
Comment (michalvasko) [inline src/main.c:1123]: If you are adding a flag, use it correctly. By default (with no options), syslog is used so I would set it to `1`. Then, for the `d` option, it should be set to `0`.

File: src/main.c [hunk 2]
```diff
@@ -1152,11 +1153,15 @@ main(int argc, char *argv[])
...
-    while ((c = getopt(argc, argv, "dhVp:f:U::m:u:g:n:i:t:x:v:c:")) != -1) {
+    while ((c = getopt(argc, argv, "dhVp:f:FU::m:u:g:n:i:t:x:v:c:")) != -1) {
         switch (c) {
         case 'd':
             daemonize = 0;
             break;
+	case 'F':
+            daemonize = 0;
+            use_syslog = 1;
+            break;
```
Comment (michalvasko) [inline src/main.c:1164]: Note that the cases are ordered the same way as the options in `getopt`, please stick to that order.

File: src/main.c [hunk 3]
```diff
@@ -1319,6 +1324,10 @@ main(int argc, char *argv[])
         np2_stderr_log = 0;
     }
 
+    /* use syslog even if running in foreground */
+    if (use_syslog == 1)
+        np2_stderr_log = 0;
```
Comment (michalvasko) [inline src/main.c:1329]: We do not allow `if` without braces in our coding style.

**Review by michalvasko — CHANGES_REQUESTED** [review]
Okay, this can be merged after some changes. Also, change the base branch to `devel`, we do not accept PRs to `master`.

**Comment by troglobit** [comment]
Will do, thanks for the swift review!

File: src/main.c (hunk 1 above)
Comment (troglobit) [reply src/main.c]: Fixed in updated PR (force-push)

File: src/main.c (hunk 3 above)
Comment (troglobit) [reply src/main.c]: Fixed in updated PR (force-push)

**Comment by troglobit** [comment]
> Okay, this can be merged after some changes. Also, change the base branch to `devel`, we do not accept PRs to `master`.

Fixed.

File: src/main.c [hunk 4]
```diff
@@ -1199,6 +1201,10 @@ main(int argc, char *argv[])
         case 'f':
             np2srv.server_dir = optarg;
             break;
+	case 'F':
+            daemonize = 0;
+            use_syslog = 1;
```
Comment (michalvasko) [inline src/main.c:1206]: Redundant now.

File: src/main.c [hunk 5]
```diff
@@ -1319,6 +1325,11 @@ main(int argc, char *argv[])
         np2_stderr_log = 0;
     }
 
+    /* use syslog even if running in foreground */
```
Comment (michalvasko) [inline src/main.c:1328]: Inaccurate now, it is not only about the `-F` parameter.

**Review by michalvasko — CHANGES_REQUESTED** [review]
Almost there but I have realized that you should probably also change the man page `doc/netopeer2-server.8`.

File: src/main.c (hunk 4 above)
Comment (troglobit) [reply src/main.c:1206]: OK

File: src/main.c [hunk 6]
```diff
@@ -1319,6 +1325,11 @@ main(int argc, char *argv[])
         np2_stderr_log = 0;
```
[the remark this answers is not in the document]
Comment (troglobit) [reply src/main.c:1325]: Argh, obviously. Sorry about that, I don't even need the `use_syslog` flag. Fixing!

File: src/main.c (hunk 5 above)
Comment (troglobit) [reply src/main.c:1328]: Fixing ...

**Comment by troglobit** [comment]
> Almost there but I have realized that you should probably also change the man page `doc/netopeer2-server.8`.

Of course, will fix that too.  New update coming later tonight (CET).

**Comment by troglobit** [comment]
There, finally.  This last changeset actually doesn't look half bad.  Note, I decided to move the case and group it closer to `-d`, which the new option has a lot in common with.  Because I noticed all other non-argument options were sorted first.

File: doc/netopeer2-server.8 [hunk 7]
```diff
@@ -8,7 +8,7 @@ netopeer2-server \- NETCONF server daemon build on libnetconf2 with sysrepo data
 .
 .SH SYNOPSIS
 .B netopeer2-server
-[\fB-dhV\fP] [\fB-p\fP \fIPATH\fP] [\fB-U\fP[\fIPATH\fP]] [\fB-m\fP \fIMODE\fP] [\fB-u\fP \fIUID\fP]
+[\fB-dFhV\fP] [\fB-p\fP \fIPATH\fP] [\fB-U\fP[\fIPATH\fP]] [\fB-m\fP \fIMODE\fP] [\fB-u\fP \fIUID\fP]
```
Comment (michalvasko) [inline doc/netopeer2-server.8:11]: Please update the last change on line 5 as well for it to be obvious that is is up-to-date.

**Review by michalvasko — CHANGES_REQUESTED** [review]
Right, code changes are really simple now so just this one last detail and it should be good to go.

File: doc/netopeer2-server.8 (hunk 7 above)
Comment (troglobit) [reply doc/netopeer2-server.8:11]: Heh, of course! :-)

**Comment by troglobit** [comment]
> Right, code changes are really simple now so just this one last detail and it should be good to go.

Aaaaand done :smiley::+1:

**Review by michalvasko — APPROVED** [review]

File: src/main.c [hunk 8]
```diff
@@ -1152,11 +1153,15 @@ main(int argc, char *argv[])
     np2srv.server_dir = SERVER_DIR;
 
     /* process command line options */
-    while ((c = getopt(argc, argv, "dhVp:f:U::m:u:g:n:i:t:x:v:c:")) != -1) {
+    while ((c = getopt(argc, argv, "dFhVp:f:U::m:u:g:n:i:t:x:v:c:")) != -1) {
         switch (c) {
         case 'd':
             daemonize = 0;
             break;
+	case 'F':
```
Comment (michalvasko) [inline src/main.c:1161]: Is this a TAB? If so, it is the reason why the tests failed.

Comment (troglobit) [reply src/main.c:1161]: Indeed, it was a TAB, really thought I had cleaned that up. Sorry, fixing!

**Pull request merged**